src/description.cpp

```diff
@@ -307,10 +307,23 @@ string Description::generateSdp(string_view eol) const {
 
 	// Entries
 	bool first = true;
+
+	std::map<std::string, std::vector<shared_ptr<Entry>>> mLines;
 	for (const auto &entry : mEntries) {
-		sdp << entry->generateSdp(eol, addr, port);
+		mLines[entry->mid()].push_back(entry);
+	}
+
+	for (const auto &mLine : mLines) {
+		std::vector<string> rids;
+		for (const auto &entry : mLine.second) {
+			if (entry->rid() != "") {
+				rids.push_back(entry->rid());
+			}
+		}
 
-		if (!entry->isRemoved() && std::exchange(first, false)) {
+		sdp << mLine.second.front()->generateSdp(eol, addr, port, rids);
```

Comment:
Does this mean that only the first entry is serialized as an m= line? For some reason I missed that when looking at the API.

I don't think this is a good idea as description entries are supposed to represent m= lines, and it creates a convoluted implementation where you gather and reinject rids. The parsing methods would also need to be updated with the same logic. 

Maybe I'm missing something, but it seems that it would be way more straightforward to implement support for rids inside `Description::Media` and use the same track for the different quality levels.

Reply:
@paullouisageneau done! Rewrote the PR to follow this design instead.